examples/server/server.cpp

```diff
@@ -13,11 +14,81 @@
 #include <string>
 #include <thread>
 #include <vector>
+#include <queue>
+#include <condition_variable>
+#include <future>
+#include <atomic>
+#include <map>
+#include <random>
+#include <iostream>
+#include <sys/stat.h>
+#if !defined(_WIN32) && !defined(_WIN64)
+#include <unistd.h>
+#else
+#include <direct.h>  // For _mkdir
+#include <io.h>      // For _access
+#ifndef W_OK
+#define W_OK 2
+#endif
+#endif
+#include <semaphore>
+#include <csignal>
+
+
+// ---- PORTABLE COUNTING SEMAPHORE ----
+class CountingSemaphore {
+public:
+    explicit CountingSemaphore(int initial) : count_(initial) {}
+    void release() {
+        std::unique_lock<std::mutex> lock(mtx_);
+        ++count_;
+        cv_.notify_one();
+    }
+    bool try_acquire() {
+        std::unique_lock<std::mutex> lock(mtx_);
+        if (count_ > 0) {
+            --count_;
+            return true;
+        }
+        return false;
+    }
+private:
+    std::mutex mtx_;
+    std::condition_variable cv_;
+    int count_;
+};
 
 using namespace httplib;
 using json = nlohmann::ordered_json;
 
-namespace {
+// Add debug logging macro for convenience
+#define SERVER_DEBUG(msg) \
+    do { \
+        if (g_debug_mode) \
+            std::cerr << "[SERVER_DEBUG] " << msg << std::endl; \
+    } while (0)
+
+// ---- GLOBALS FOR SIGNAL HANDLING ----
+std::atomic<bool> shutdown_requested{false};
+httplib::Server *g_svr_ptr = nullptr;
+std::atomic<int> g_active_tasks{0};
+std::unique_ptr<CountingSemaphore> task_queue_slots;
+std::atomic<int> active_http_requests{0};
+// --------------------------------------
+
+bool g_debug_mode = false;
+
+// RAII guard for HTTP request counting
+struct HttpRequestCounter {
+    explicit HttpRequestCounter(std::atomic<int> &counter)
+        : ctr(counter) {
+        ctr.fetch_add(1, std::memory_order_relaxed);
+    }
+    ~HttpRequestCounter() {
+        ctr.fetch_sub(1, std::memory_order_relaxed);
+    }
+    std::atomic<int>& ctr;
+};
```

Comment:
These are too much synchronization primitives, which makes me think that there are likely issues with the synchronization logic. Most of these should not be needed. A simple task queue with a mutex should be enough for the server logic.